lib/matplotlib/figure.py

```diff
@@ -2739,6 +2739,13 @@ def draw(self, renderer):
 
         self.canvas.draw_event(renderer)
 
+    def draw_no_output(self):
+        """
+        Draw the figure with no output.  Useful to get the final size of
+        artists that require a draw before their size is known (e.g. text).
+        """
+        _no_output_draw(self)
```

Comment:
If we are going to pull this up is it better to move the implementation here and replace our internal use of `_no_output_draw` with `fig.draw_no_output()`?

Reply:
Done - lets see if I made any mistakes via CI